File: redu/core/src/main/java/br/com/unibh/redu/core/enums/EstadoDoacaoEnum.java
package br.com.unibh.redu.core.enums;

public enum EstadoDoacaoEnum {

	NOVO("Novo"),
	USADO("Usado"),
	COM_DEFEITO("Com defeito");
	
	private final String descricao;
	
	private EstadoDoacaoEnum(String descricao) {
		this.descricao =descricao;
	}
	
	@Override
	public String toString() {
		return descricao;
	}
}

File: redu/web/src/app/modules/shared/resource/enum/estado-doacao.enum.ts
export enum EstadoDoacaoEnum {
    NOVO = 'Novo',
    USADO = 'Usado',
    COM_DEFEITO = 'Com defeito',
}

export namespace EstadoDoacaoEnum {

    export function valueOf(value: string) {
        switch (value) {
            case 'NOVO':
                return EstadoDoacaoEnum.NOVO;
            case 'USADO':
                return EstadoDoacaoEnum.USADO;
            case 'COM_DEFEITO':
                return EstadoDoacaoEnum.COM_DEFEITO;
        }
    }
}
